### PR TITLE
add nginx access log toggle

### PR DESCRIPTION
**Pull request by TuTuRyYE**
**What this PR does**:
Add a toggle to enable/disable access logs in nginx (similar to loki helm chart)

**Which issue(s) this PR fixes**:
Nothing

**Checklist**
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: templates/nginx/nginx-config.yaml [hunk 1]
```diff
@@ -28,7 +28,9 @@ data:
       log_format   main '$remote_addr - $remote_user [$time_local]  $status '
         '"$request" $body_bytes_sent "$http_referer" '
         '"$http_user_agent" "$http_x_forwarded_for" $http_x_scope_orgid';
+      {{- if .Values.nginx.config.verboseLogging }}
       access_log   /dev/stderr  main;
+      {{- else }}
```
Comment (nschad) [inline templates/nginx/nginx-config.yaml:33]:
```suggestion
      {{- end }}
```

Should this be `end` ?

**Review by nschad — CHANGES_REQUESTED** [review]
Thank you. However you still need to update CHANGELOG.md with your PR! :)

Also you must sign your commits with DCO!

**Comment by TuTuRyYE** [comment]
I'm not used to contributing, this is my first time. Thanks for your help :)

File: templates/nginx/nginx-config.yaml [hunk 2]
```diff
@@ -28,7 +28,9 @@ data:
       log_format   main '$remote_addr - $remote_user [$time_local]  $status '
         '"$request" $body_bytes_sent "$http_referer" '
         '"$http_user_agent" "$http_x_forwarded_for" $http_x_scope_orgid';
+      {{- if .Values.nginx.config.verboseLogging }}
```
Comment (nschad) [inline templates/nginx/nginx-config.yaml:31]: maybe we should rename this `accessLogging`. Verbose logging could be a number of things

**Comment by nschad** [comment]
> I'm not used to contributing, this is my first time. Thanks for your help :)

Absolutely not a problem. Thank you for your contribution

File: README.md [hunk 3]
```diff
@@ -490,6 +490,7 @@ Kubernetes: `^1.19.0-0`
 | nginx.&ZeroWidthSpace;config.&ZeroWidthSpace;client_max_body_size | string | `"1M"` | ref: http://nginx.org/en/docs/http/ngx_http_core_module.html#client_max_body_size |
 | nginx.&ZeroWidthSpace;config.&ZeroWidthSpace;dnsResolver | string | `"kube-dns.kube-system.svc.cluster.local"` |  |
 | nginx.&ZeroWidthSpace;config.&ZeroWidthSpace;dnsTTL | string | `"15s"` | (optional) Including the valid parameter to the `resolver` directive to re-resolve names every `dnsTTL` seconds/minutes |
+| nginx.&ZeroWidthSpace;config.&ZeroWidthSpace;verboseLogging | bool | `true` | Enable access log to nginx |
```
[the remark this answers is not in the document]
Comment (TuTuRyYE) [reply README.md]: I didn't know about this tool, thank you very much

File: templates/nginx/nginx-config.yaml (hunk 2 above)
Comment (TuTuRyYE) [reply templates/nginx/nginx-config.yaml]: Actually after talking with my team, we would like only to ignore 2XX and 3XX status code request in the access logs.
I made the change in that direction, should I do some change on the PR description ? or should I create a new one ?

**Review by nschad — CHANGES_REQUESTED** [review]

File: templates/nginx/nginx-config.yaml (hunk 2 above)
Comment (nschad) [reply templates/nginx/nginx-config.yaml]: Nope this PR is fine

File: README.md [hunk 4]
```diff
@@ -485,15 +485,16 @@ Kubernetes: `^1.19.0-0`
...
 | nginx.&ZeroWidthSpace;autoscaling.&ZeroWidthSpace;targetMemoryUtilizationPercentage | int | `0` |  |
-| nginx.&ZeroWidthSpace;config.&ZeroWidthSpace;auth_orgs | list | `[]` | (optional) List of [auth tenants](https://cortexmetrics.io/docs/guides/auth/) to set in the nginx config |
-| nginx.&ZeroWidthSpace;config.&ZeroWidthSpace;basicAuthSecretName | string | `""` | (optional) Name of basic auth secret. In order to use this option, a secret with htpasswd formatted contents at the key ".htpasswd" must exist. For example:   apiVersion: v1   kind: Secret   metadata:     name: my-secret     namespace: <same as cortex installation>   stringData:     .htpasswd: |       user1:$apr1$/woC1jnP$KAh0SsVn5qeSMjTtn0E9Q0       user2:$apr1$QdR8fNLT$vbCEEzDj7LyqCMyNpSoBh/ Please note that the use of basic auth will not identify organizations the way X-Scope-OrgID does. Thus, the use of basic auth alone will not prevent one tenant from viewing the metrics of another. To ensure tenants are scoped appropriately, explicitly set the `X-Scope-OrgID` header in the nginx config. Example   setHeaders:     X-Scope-OrgID: $remote_user |
```
Comment (nschad) [inline README.md:489]: This diff shouldn't be here. What helm-docs version are you using? The newest or v1.11.0?

I will also update that in our pipeline, our version is quite old

Comment (nschad) [reply README.md:489]: Probably is my old version though.

Comment (nschad) [reply README.md:489]: Yep this looks like our old version is at fault. Created a PR here: #404

File: templates/nginx/nginx-config.yaml [hunk 5]
```diff
@@ -28,7 +28,18 @@ data:
       log_format   main '$remote_addr - $remote_user [$time_local]  $status '
         '"$request" $body_bytes_sent "$http_referer" '
         '"$http_user_agent" "$http_x_forwarded_for" $http_x_scope_orgid';
+      
+      {{- if .Values.gateway.verboseLogging }}
```
[the remark this answers is not in the document]
Comment (TuTuRyYE) [reply templates/nginx/nginx-config.yaml]: i copy/paste it from loki helm chart ...
I changed it sry

**Review by nschad — APPROVED** [review]

**Review by kd7lxl — APPROVED** [review]

**Comment by nschad** [comment]
Hey @TuTuRyYE,

I shouldn't have pressed the GitHub "update-repo" button. What a mess. Anyway... DCO seems to be stuck.
Can you do the following on your PC in your cortex-helm-chart repo?

_UNLESS YOU HAVE LOCAL CHANGES IN THAT BRANCH_

```
git fetch origin add_logging_verbose_option_to_nginx
git reset --hard origin/add_logging_verbose_option_to_nginx
# e2de2f1f3dd1ecb95762851cf96c512027b71e8f should be last commit ID from master. Your branch is already rebased. This should clean up any potential weird authors.
git rebase -i e2de2f1f3dd1ecb95762851cf96c512027b71e8f -x "git commit --amend --reset-author -CHEAD"
# push it
git push -u origin add_logging_verbose_option_to_nginx -f
```

**Pull request merged**